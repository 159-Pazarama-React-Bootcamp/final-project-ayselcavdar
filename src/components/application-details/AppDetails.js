import {  useState } from 'react';
import styles from './AppDetails.module.css';
import Button from '../button/Button';
import { useNavigate, useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import CommentBox from '../comment/CommentBox';
import { editApplication } from '../../redux/actions/crudActions';
import { useDispatch, useSelector } from 'react-redux';
import { n, statusEnum } from '../../constants/enum';

const AppDetails = ({ isAdmin }) => {
  const [comment, setComment] = useState('');
  const dispatch = useDispatch();
  const { basvuruNo } = useParams();
  const { applications } = useSelector((state) => state?.data);
  const details = applications?.find((item) => item.id == basvuruNo);

  const handleComment = (e) => setComment(e.target.value);

  const submitCommentLine = () => {
    const data = {
      ...details,
      hasResponse:true,
      responseText:comment,
      status: statusEnum.ANSWERED.name,
    };
    if(comment.length > 3){
        dispatch(editApplication(data));
        setComment('');
    }
    return false;
  };

  const navigate = useNavigate();

  return (
    <div className={styles['details-container']}>
      <div className={styles['subdetails-container']}>
        <div>
          <h3 className={styles['detail-title']}>
            {details?.id.slice(-n).toLowerCase()} numaralı başvuru bilgileriniz.
          </h3>
          <div>
            <ul className={styles['detail-items']}>
              <li>
                <span style={{ fontWeight: '500', fontSize: '1.2rem' }}>
                  Ad-Soyad:
                </span>{' '}
                {details?.firstName} {details?.lastName}
              </li>
              <li>
                <span className={styles['detail-item--label']}>TC Num:</span>{' '}
                {details?.tcNum}
              </li>
              <li>
                <span className={styles['detail-item--label']}>Yaş:</span>{' '}
                {details?.age}
              </li>
              <li>
                <span className={styles['detail-item--label']}>
                  Başvuru nedeni:
                </span>{' '}
                {details?.reason}
              </li>
              <li>
                <span className={styles['detail-item--label']}>Address:</span>{' '}
                {details?.address}
              </li>
              <li>
                <span className={styles['detail-item--label']}>
                  Başvuru Durumu:
                </span>{' '}
                {statusEnum[details?.status].descr}.
              </li>
              {isAdmin && !details?.hasResponse && (
                <CommentBox
                  comment={comment}
                  handleComment={handleComment}
                  submitCommentLine={submitCommentLine}
                />
              )}
              {details?.responseText && (
                <li>
                  <span className={styles['detail-item--label']}>
                    Cevap:
                  </span>{' '}
                  {details?.responseText}.
                </li>
              )}
            </ul>
          </div>
          <Button
            type="button"
            content={isAdmin ? 'Başvurular' : 'Başvurularım'}
            onClick={() => navigate(-1)}
          />
        </div>
        <div>
          {details?.photo ? (
            <img
              className={styles['avatar']}
              src={details?.photo}
              alt={`photo of ${details?.name}`}
            />
          ) : (
            <img
              className={styles['avatar']}
              src="https://www.w3schools.com/howto/img_avatar2.png"
            />
          )}
        </div>
      </div>
    </div>
  );
};

export default AppDetails;

AppDetails.propTypes = {
  isAdmin: PropTypes.bool,
};
