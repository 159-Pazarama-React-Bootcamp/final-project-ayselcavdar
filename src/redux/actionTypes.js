const LOGIN_START = 'LOGIN_START';
const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
const LOGIN_FAIL = 'LOGIN_FAIL';

const LOGOUT_START = 'LOGOUT_START';
const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
const LOGOUT_FAIL = 'LOGOUT_FAIL';

const SET_USER = 'SET_USER';

const GET_APPLICATIONS_START = 'GET_APPLICATIONS_START';
const GET_APPLICATIONS_SUCCESS = 'GET_APPLICATIONS_SUCCESS';
const GET_APPLICATIONS_FAIL = 'GET_APPLICATIONS_FAIL';

const DELETE_APPLICATION_START = 'DELETE_APPLICATION_START';
const DELETE_APPLICATION_FAIL = 'DELETE_APPLICATION_FAIL';
const DELETE_APPLICATION_SUCCESS = 'DELETE_APPLICATION_SUCCESS';

const CREATE_APPLICATION_START = 'CREATE_APPLICATION_START';
const CREATE_APPLICATION_SUCCESS = 'CREATE_APPLICATION_SUCCESS';
const CREATE_APPLICATION_FAIL = 'CREATE_APPLICATION_FAIL';

const EDIT_APPLICATION_START = 'EDIT_APPLICATION_START';
const EDIT_APPLICATION_FAIL = 'EDIT_APPLICATION_FAIL';
const EDIT_APPLICATION_SUCCESS = 'EDIT_APPLICATION_SUCCESS';


export {
  LOGIN_START,
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGOUT_START,
  LOGOUT_SUCCESS,
  LOGOUT_FAIL,
  SET_USER,
  CREATE_APPLICATION_FAIL,
  CREATE_APPLICATION_START,
  CREATE_APPLICATION_SUCCESS,
  GET_APPLICATIONS_FAIL,
  GET_APPLICATIONS_START,
  GET_APPLICATIONS_SUCCESS,
  DELETE_APPLICATION_START,
  DELETE_APPLICATION_SUCCESS,
  DELETE_APPLICATION_FAIL,
  EDIT_APPLICATION_FAIL,
  EDIT_APPLICATION_START,
  EDIT_APPLICATION_SUCCESS,
};
