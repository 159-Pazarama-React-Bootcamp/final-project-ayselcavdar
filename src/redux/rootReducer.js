import {combineReducers} from 'redux';
import { authReducer } from './reducers/authReducer';
import applicationReducer from './reducers/applicationReducer';

const reducers = {
  user: authReducer,
  data: applicationReducer,
};

export const rootReducer = combineReducers(reducers);