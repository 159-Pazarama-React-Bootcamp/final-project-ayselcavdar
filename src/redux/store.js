import {createStore, applyMiddleware} from 'redux';
import {rootReducer} from './rootReducer';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const middleware = [thunk];
// state değişikliklerini takibe yarayan anlık loglar sunan logger modulü,
// yalnızca development ortamı için uygulanmıştır
if(process.env.NODE_ENV === 'development'){
    middleware.push(logger);
}
export const store = createStore(rootReducer, applyMiddleware(...middleware));