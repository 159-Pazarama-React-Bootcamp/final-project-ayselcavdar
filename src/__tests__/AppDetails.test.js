/**
 * @jest-environment jsdom
 */
import render from 'react-test-renderer';
import AppDetails from '../components/application-details/AppDetails';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import { BrowserRouter } from 'react-router-dom';

describe('<AppDetails /> rendering', () => {
  it('AppDetails snapshots check', () => {
    const wrapper = render.create(
      <Provider store={store}>
        <BrowserRouter>
          <AppDetails isAdmin />
        </BrowserRouter>
      </Provider>
    );
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
