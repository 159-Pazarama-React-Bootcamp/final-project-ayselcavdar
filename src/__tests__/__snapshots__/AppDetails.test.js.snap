// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppDetails /> rendering AppDetails render check 1`] = `
<div
  className="details-container"
>
  <div
    className="subdetails-container"
  >
    <div>
      <h3
        className="detail-title"
      >
         numaralı başvuru
         
        bilgileri
        .
      </h3>
      <div>
        <ul
          className="detail-items"
        >
          <li>
            <span
              style={
                Object {
                  "fontSize": "1.2rem",
                  "fontWeight": "500",
                }
              }
            >
              Ad-Soyad:
            </span>
             
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              TC Num:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Yaş:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Başvuru nedeni:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Address:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Başvuru Durumu:
            </span>
             
            .
          </li>
          <div
            className="comment-box"
          >
            <textarea
              cols="35"
              onChange={[Function]}
              placeholder="başvuruyu cevapla"
              rows="2"
              value=""
            />
            <span
              className="disable"
              onClick={[Function]}
            >
              Gönder
            </span>
          </div>
        </ul>
      </div>
      <button
        className="btn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Başvurular
      </button>
    </div>
    <div>
      <img
        alt="photo of undefined"
        className="avatar"
        src="https://www.w3schools.com/howto/img_avatar2.png"
      />
    </div>
  </div>
</div>
`;

exports[`<AppDetails /> rendering AppDetails snapshots check 1`] = `
<div
  className="details-container"
>
  <div
    className="subdetails-container"
  >
    <div>
      <h3
        className="detail-title"
      >
         numaralı başvuru
         
        bilgileri
        .
      </h3>
      <div>
        <ul
          className="detail-items"
        >
          <li>
            <span
              style={
                Object {
                  "fontSize": "1.2rem",
                  "fontWeight": "500",
                }
              }
            >
              Ad-Soyad:
            </span>
             
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              TC Num:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Yaş:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Başvuru nedeni:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Address:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Başvuru Durumu:
            </span>
             
            .
          </li>
          <div
            className="comment-box"
          >
            <textarea
              cols="35"
              onChange={[Function]}
              placeholder="başvuruyu cevapla"
              rows="2"
              value=""
            />
            <span
              className="disable"
              onClick={[Function]}
            >
              Gönder
            </span>
          </div>
        </ul>
      </div>
      <button
        className="btn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Başvurular
      </button>
    </div>
    <div>
      <img
        alt="photo of undefined"
        className="avatar"
        src="https://www.w3schools.com/howto/img_avatar2.png"
      />
    </div>
  </div>
</div>
`;

exports[`AppDetails render check 1`] = `
<div
  className="details-container"
>
  <div
    className="subdetails-container"
  >
    <div>
      <h3
        className="detail-title"
      >
         numaralı başvuru
         
        bilgileri
        .
      </h3>
      <div>
        <ul
          className="detail-items"
        >
          <li>
            <span
              style={
                Object {
                  "fontSize": "1.2rem",
                  "fontWeight": "500",
                }
              }
            >
              Ad-Soyad:
            </span>
             
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              TC Num:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Yaş:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Başvuru nedeni:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Address:
            </span>
             
          </li>
          <li>
            <span
              className="detail-item--label"
            >
              Başvuru Durumu:
            </span>
             
            .
          </li>
          <div
            className="comment-box"
          >
            <textarea
              cols="35"
              onChange={[Function]}
              placeholder="başvuruyu cevapla"
              rows="2"
              value=""
            />
            <span
              className="disable"
              onClick={[Function]}
            >
              Gönder
            </span>
          </div>
        </ul>
      </div>
      <button
        className="btn"
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        Başvurular
      </button>
    </div>
    <div>
      <img
        alt="photo of undefined"
        className="avatar"
        src="https://www.w3schools.com/howto/img_avatar2.png"
      />
    </div>
  </div>
</div>
`;
